plugins/outputs/mqtt/mqtt_v5.go

```diff
@@ -13,13 +13,57 @@ import (
 	"github.com/influxdata/telegraf/internal"
 )
 
+// mqtt v5-specific publish properties.
+// See https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901109
+type mqttv5PublishProperties struct {
+	ContentType    string            `toml:"content_type"`
+	ResponseTopic  string            `toml:"response_topic"`
+	MessageExpiry  config.Duration   `toml:"message_expiry"`
+	TopicAlias     *uint16           `toml:"topic_alias"`
+	UserProperties map[string]string `toml:"user_properties"`
+}
+
 type mqttv5Client struct {
 	*MQTT
-	client *mqttv5auto.ConnectionManager
+	client            *mqttv5auto.ConnectionManager
+	publishProperties *mqttv5.PublishProperties
 }
 
 func newMQTTv5Client(cfg *MQTT) *mqttv5Client {
-	return &mqttv5Client{MQTT: cfg}
+	return &mqttv5Client{
+		MQTT:              cfg,
+		publishProperties: buildPublishProperties(cfg),
+	}
+}
+
+// Build the v5 specific publish properties if they are present in the
+// config.
+// These should not change during the lifecycle of the client.
+func buildPublishProperties(cfg *MQTT) *mqttv5.PublishProperties {
+	cfgProps := cfg.V5PublishProperties
```

Comment:
Any reason to declare a new variable? Why not verify `cfg.V5PublishProperties` is nil and use that below?